helm_deploy/dashboards/grafana-sidekiq-metrics.yml

```diff
@@ -0,0 +1,910 @@
+  labels:
+    grafana_dashboard: ""
+data:
+  grafana-sidekiq-metrics.json: |
+    {
+      "annotations": {
+        "list": [
+          {
+            "builtIn": 1,
+            "datasource": {
+              "type": "datasource",
+              "uid": "grafana"
+            },
+            "enable": true,
+            "hide": true,
+            "iconColor": "rgba(0, 211, 255, 1)",
+            "name": "Annotations & Alerts",
+            "type": "dashboard"
+          }
+        ]
+      },
+      "description": "Visual Sidekiq Metrics",
+      "editable": true,
+      "fiscalYearStartMonth": 0,
+      "graphTooltip": 0,
+      "id": 166,
+      "links": [],
+      "liveNow": false,
+      "panels": [
+        {
+          "datasource": {
+            "type": "prometheus",
+            "uid": "prometheus"
+          },
+          "description": "",
+          "fieldConfig": {
+            "defaults": {
+              "color": {
+                "mode": "thresholds"
+              },
+              "mappings": [
+                {
+                  "options": {
+                    "match": "null",
+                    "result": {
+                      "text": "N/A"
+                    }
+                  },
+                  "type": "special"
+                }
+              ],
+              "thresholds": {
+                "mode": "absolute",
+                "steps": [
+                  {
+                    "color": "#299c46",
+                    "value": null
+                  },
+                  {
+                    "color": "rgba(237, 129, 40, 0.89)",
+                    "value": 1
+                  },
+                  {
+                    "color": "#d44a3a",
+                    "value": 10
+                  }
+                ]
+              },
+              "unit": "none",
+              "unitScale": true
+            },
+            "overrides": []
+          },
+          "gridPos": {
+            "h": 4,
+            "w": 3,
+            "x": 0,
+            "y": 0
+          },
+          "id": 10,
+          "links": [],
+          "maxDataPoints": 100,
+          "options": {
+            "colorMode": "value",
+            "graphMode": "area",
+            "justifyMode": "auto",
+            "orientation": "horizontal",
+            "reduceOptions": {
+              "calcs": [
+                "lastNotNull"
+              ],
+              "fields": "",
+              "values": false
+            },
+            "showPercentChange": false,
+            "textMode": "auto",
+            "wideLayout": true
+          },
+          "pluginVersion": "10.3.3",
+          "targets": [
+            {
+              "datasource": {
+                "type": "prometheus",
+                "uid": "prometheus"
+              },
+              "editorMode": "code",
+              "expr": "avg(ruby_sidekiq_stats_processed{namespace=\"$namespace\",pod=~\"laa-.+-app-.+\"})",
```

Comment:
This is telling me that in production we've processed nearly 40,000 jobs in the app store. Which I don't think I believe, as we only use jobs for quite specific events that don't happen that often. I wonder if this metric means something different to what we think it means?

Reply:
This is "all time" and the high numbers should match what sidekiq webui is displaying for "processed". on assess and submit production envs this should be 66k+ and 63k+. On app store you can use rails console

```
Sidekiq::Stats.new.fetch_stats!
=> {:processed=>39259, :failed=>14, :scheduled_size=>2, :retry_size=>0, :dead_size=>2, :processes_size=>2, :default_queue_latency=>0, :workers_size=>0, :enqueued=>0}
```

Reply:
But the only jobs the production app store should have processed is NotifySubscriber jobs, which should be some multiple of the number of SubmissionVersion records, which number in the hundreds. What _are_ the other jobs?

Reply:
So SidekiqWorker::Alive has been running since we switched to this API 3 months ago - this is likely the source of the high numbers. But if I'm correct, sidekiq alive is run every 30 seconds (the liveness probe has a period of 30 seconds) if this is the case, the number of jobs executed should be something more in the hundreds of thousands if we assume the worker has been live for most of the 3 months - so it's all a bit mystifying. Honestly though the real fact is that this stat alone doesn't really help us much